src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -91,6 +93,18 @@ public static IHostBuilder AddWebScriptHost(this IHostBuilder builder, IServiceP
                     loggingBuilder.AddWebJobsSystem<SystemLoggerProvider>();
                     if (environment.IsAzureMonitorEnabled())
                     {
+                        if (environment.IsConsumptionOnLegion())
+                        {
+                            loggingBuilder.Services.AddOptions<LoggerFilterOptions>()
+                                .Configure<IOptionsMonitor<AppServiceOptions>>((filters, options) =>
+                                {
+                                    bool isAzureMonitorLoggingEnabled = options.CurrentValue.AzureMonitorLoggingEnabled;
+                                    filters.AddFilter<AzureMonitorDiagnosticLoggerProvider>((category, level) =>
+                                    {
+                                        return isAzureMonitorLoggingEnabled;
```

Comment:
Small nit: recommend not capturing `isAzureMonitorLoggingEnabled` into a bool. Just to avoid any potential inconsistencies with options and filter refresh.

```suggestion
                                        return options.CurrentValue.AzureMonitorLoggingEnabled;
```

Reply:
Updated